smithy-typescript-codegen/src/test/java/software/amazon/smithy/typescript/codegen/endpointsV2/RuleSetParameterFinderTest.java

```diff
@@ -0,0 +1,114 @@
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.rulesengine.language.EndpointRuleSet;
+import software.amazon.smithy.rulesengine.traits.EndpointRuleSetTrait;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class RuleSetParameterFinderTest {
+
+    Node ruleSet = Node.parse("""
+            {
+              "version": "1.0",
+              "parameters": {
+                "BasicParameter": {
+                  "required": false,
+                  "documentation": "...",
+                  "type": "String"
+                },
+                "NestedParameter": {
+                  "required": true,
+                  "documentation": "...",
+                  "type": "Boolean"
+                },
+                "UrlOnlyParameter": {
+                  "required": true,
+                  "documentation": "...",
+                  "type": "String"
+                },
+                "UnusedParameter": {
+                  "required": false,
+                  "documentation": "...",
+                  "type": "String"
+                }
+              },
+              "rules": [
+                {
+                  "conditions": [
+                    {
+                      "fn": "isSet",
+                      "argv": [
+                        {
+                          "ref": "BasicParameter"
+                        }
+                      ]
+                    }
+                  ],
+                  "rules": [
+                    {
+                      "conditions": [
+                        {
+                          "fn": "booleanEquals",
+                          "argv": [
+                            {
+                              "fn": "booleanEquals",
+                              "argv": [
+                                {
+                                  "fn": "booleanEquals",
+                                  "argv": [
+                                    {
+                                      "fn": "booleanEquals",
+                                      "argv": [
+                                        {
+                                          "ref": "NestedParameter"
+                                        },
+                                        true
+                                      ]
+                                    },
+                                    true
+                                  ]
+                                },
+                                true
+                              ]
+                            },
+                            true
+                          ]
+                        }
+                      ],
+                      "endpoint": {
+                        "url": "https://www.{BasicParameter}.{UrlOnlyParameter}.com",
```

Comment:
Could you include a test with a more complex url, maybe with a [template shorthand](https://smithy.io/2.0/additional-specs/rules-engine/specification.html#template-shorthand), as a sanity check on the regex and to make sure we don't break it down the road?

Reply:
added test case for shorthand, which I didn't know existed

Reply:
I mean one so that it has the `#` in the url, sorry. Just so we make sure those kinds of urls are matched by the regex.

Reply:
Nevermind, practically speaking we're only looking for endpoint param names, so the object path won't matter.